src/formatter.rs

```diff
@@ -158,3 +159,44 @@ pub fn print_call(call: &Call, padding: usize, show_calls: &ShowCalls, resolve_h
         print_call(subcall, padding + 2, show_calls, resolve_hashes);
     }
 }
+
+pub fn print_logs(log_query: &StorageLogQuery) {
+    println!("\nSTORAGE LOGS");
+    println!("------------");
+    let separator = "-".repeat(80);
+    println!("{}", separator);
+    println!("{:<15} {:?}", "Type:", log_query.log_type);
+    println!("{:<15} {}", "Address:", log_query.log_query.address);
+    println!(
+        "{:<15} {}",
+        "Key:",
+        format!("{:?}", log_query.log_query.key)
+    );
+    println!(
+        "{:<15} {}",
+        "Read Value:",
+        format!("{:?}", log_query.log_query.read_value)
+    );
+    println!(
+        "{:<15} {}",
+        "Written Value:",
+        format!("{:?}", log_query.log_query.written_value)
+    );
+    println!("{}", separator);
+}
+
+pub fn print_vm_details(result: &VmPartialExecutionResult) {
+    println!("\n+---------------------+");
+    println!("| VM EXECUTION RESULTS |");
+    println!("+---------------------+\n");
+
+    println!("Cycles Used:          {}", result.cycles_used);
+    println!("Computation Gas Used: {}", result.computational_gas_used);
+    println!("Contracts Used:       {}", result.contracts_used);
+
+    if let Some(revert_reason) = &result.revert_reason {
+        println!("\n[!] Revert Reason:    {}", revert_reason);
```

Comment:
If there's a `revert_reason`, we should consider printing this on a red background to highlight that something unexpected happened:
```suggestion
        println!(format!("\n[!] Revert Reason:    {}", revert_reason).on_red());
```